tests/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/token_test.yml

```diff
@@ -25,6 +28,32 @@
         msg: 'unexpected secret values'
       when: kv2_secret1['value'] != 'foo1' or kv2_secret2['value'] != 'foo2'
 
+    - name: 'Check kv2 secret raw return value'
+      fail:
+        msg:
+      when: >-
+        'data' not in kv2_secret2_as_raw
+        or 'data' not in kv2_secret2_as_raw['data']
+        or 'metadata' not in kv2_secret2_as_raw['data']
+
+    - name: "Check multiple secrets as dict"
+      fail:
+        msg: 'Return value was not dict or items do not match.'
+      when: (kv2_secrets_as_dict | type_debug != 'dict') or (kv2_secrets_as_dict['value{{ item }}'] != 'foo{{ item }}')
```

Comment:
@briantist I missed this when I was skimming through this PR before, but noticed it when I was updating the tests. For future reference, things like this should be written using concatenation instead of relying on double templating. Conditionals like `when` are implicitly templated, so putting `{{ }}` in them 'works' but is rarely necessary and can end up doing confusing things.

```
when: (kv2_secrets_as_dict | type_debug != 'dict') or (kv2_secrets_as_dict['value' ~ item ] != 'foo' ~ item)
```

Reply:
Thanks @flowerysong I'm aware of that conditionals are already templated, though for the life of me I can't remember why I used templating there. Will keep it mind!